### PR TITLE
Add support for interface implementing other interfaces

### PR DESCRIPTION
**Pull request by maartenvanvliet**
Fixes https://github.com/absinthe-graphql/absinthe/issues/990

Adds support for interfaces implementing other interfaces according to the spec https://spec.graphql.org/draft/#sec-Interfaces.Interfaces-Implementing-Interfaces

 * added support in the parser/sdl schema
 * added support in macro schema
 * check if interfaces don't form a cycle
 * add support for rendering in the sdl

### CODE REVIEW

**Review by benwilson512 — APPROVED** [review]
Looks great, thank you!

File: test/support/fixtures/dynamic/interface_cycle_schema.exs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  interface Named implements Node & Named {
+    id: ID!
+    name: String
+  }
+  """
+
+  import_sdl @sdl
+  # query do
```
Comment (binaryseed) [inline test/support/fixtures/dynamic/interface_cycle_schema.exs:25]: Leftover lines?

**Review by binaryseed — COMMENTED** [review]
This is great!

File: test/support/fixtures/dynamic/interface_cycle_schema.exs (hunk 1 above)
Comment (maartenvanvliet) [reply test/support/fixtures/dynamic/interface_cycle_schema.exs:25]: yes, let me fix that

Comment (maartenvanvliet) [reply test/support/fixtures/dynamic/interface_cycle_schema.exs:25]: https://github.com/absinthe-graphql/absinthe/pull/1013

**Pull request merged**